lending_pool/tests/rust_tests.rs

```diff
@@ -125,3 +214,132 @@ fn borrow_test() {
+        lending_pool::contract_obj,
+        liquidity_pool::contract_obj,
+        aggregator_mock::contract_obj,
+    );
+    let user_addr = lending_setup.first_user_addr.clone();
+
+    lending_setup
+        .b_mock
+        .set_esdt_balance(&user_addr, USDC_TOKEN_ID, &rust_biguint!(1_000));
+
+    lending_setup
+        .b_mock
+        .execute_esdt_transfer(
+            &user_addr,
+            &lending_setup.liquidity_pool_wrapper,
+            USDC_TOKEN_ID,
+            0,
+            &rust_biguint!(1_000),
+            |sc| {
+                sc.deposit_asset(managed_address!(&user_addr));
+            },
+        )
+        .assert_ok();
+
+    lending_setup.b_mock.check_nft_balance(
+        &user_addr,
+        LEND_USDC_TOKEN_ID,
+        1,
+        &rust_biguint!(1_000),
+        &(),
+    );
+
+    lending_setup.b_mock.set_block_timestamp(1000);
+
+    lending_setup
+        .b_mock
+        .set_nft_balance(&user_addr, LEND_EGLD, 1, &rust_biguint!(5), &());
+
+    lending_setup
+        .b_mock
+        .execute_esdt_transfer(
+            &user_addr,
+            &lending_setup.liquidity_pool_wrapper,
+            LEND_EGLD,
+            1,
+            &rust_biguint!(5),
+            |sc| {
+                sc.borrow(
+                    managed_address!(&user_addr),
+                    TokenAmountPair::new(managed_token_id!(EGLD_TOKEN_ID), 0, managed_biguint!(5)),
+                    managed_biguint!(500_000_000),
+                );
+            },
+        )
+        .assert_ok();
+
+    lending_setup.b_mock.check_nft_balance(
+        &user_addr,
+        BORROW_USDC_TOKEN_ID,
+        1,
+        &rust_biguint!(500),
+        &Vec::<u8>::new(),
+    );
+
+    lending_setup.b_mock.check_nft_balance(
+        &user_addr,
+        USDC_TOKEN_ID,
+        0,
+        &rust_biguint!(500),
+        &Vec::<u8>::new(),
+    );
+
+    //lending_setup.b_mock.set_block_epoch(100);
+    lending_setup.b_mock.set_block_timestamp(2_000);
+
+    let mut payments = Vec::with_capacity(2);
+
+    payments.push(TxInputESDT {
+        token_identifier: BORROW_USDC_TOKEN_ID.to_vec(),
+        nonce: 1,
+        value: rust_biguint!(500),
+    });
+
+    payments.push(TxInputESDT {
+        token_identifier: USDC_TOKEN_ID.to_vec(),
+        nonce: 0,
+        value: rust_biguint!(500),
+    });
+
+    lending_setup
+        .b_mock
+        .execute_esdt_multi_transfer(
+            &user_addr,
+            &lending_setup.liquidity_pool_wrapper,
+            &payments,
+            |sc| {
+                sc.repay(managed_address!(&user_addr));
+            },
+        )
+        .assert_ok();
+
+    lending_setup.b_mock.check_nft_balance(
+        &user_addr,
+        BORROW_USDC_TOKEN_ID,
+        1,
+        &rust_biguint!(0),
+        &Vec::<u8>::new(),
+    );
+
+    lending_setup.b_mock.check_nft_balance(
+        &user_addr,
+        USDC_TOKEN_ID,
+        0,
+        &rust_biguint!(0),
+        &Vec::<u8>::new(),
+    );
```

Comment:
This seems wrong. The user should have to pay some additional debt when repaying after borrow. In the current test, it seems the user only pays the same amount back, with no extra for debt.

Reply:
All the scenarios are atemporal - happening instantly. The same issue happens with the deposit also.
I will change the scenario to add time so that the borrower will pay fees and the lender will receive fees.

Reply:
@dorin-iancu , fixed this. Now the borrower sends more tokens (borrowed position + accumulated_debt)